numpy/core/src/multiarray/ctors.c

```diff
@@ -1912,10 +1912,21 @@ PyArray_GetArrayParamsFromObject(PyObject *op,
 NPY_NO_EXPORT PyObject *
 PyArray_FromAny(PyObject *op, PyArray_Descr *newtype, int min_depth,
                 int max_depth, int flags, PyObject *context)
+{
+    return PyArray_FromAny_MaxDim(op, newtype, NPY_MAXDIMS, min_depth, max_depth,
+                            flags, context);
+}
+
+/*NUMPY_API
+ * Does not check for NPY_ARRAY_ENSURECOPY and NPY_ARRAY_NOTSWAPPED in flags
+ * Steals a reference to newtype --- which can be NULL
+ */
+NPY_NO_EXPORT PyObject *
+PyArray_FromAny_MaxDim(PyObject *op, PyArray_Descr *newtype, int maxdims,
+                int min_depth, int max_depth, int flags, PyObject *context)
 {
     /*
-     * This is the main code to make a NumPy array from a Python
-     * Object.  It is called from many different places.
+     * This is a duplication of PyArray_FromAny to support the maxdims option. 
      */
```

Comment:
I think you probably want to put back the old comment.

Reply:
Fixed